### PR TITLE
GEOPY-382

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-382 - StatsCache remains when property values are edited through geoh5py**

### CODE REVIEW

File: geoh5py/io/h5_writer.py [hunk 1]
```diff
@@ -559,6 +559,10 @@ def write_data_values(
                     compression="gzip",
                     compression_opts=9,
                 )
+                entity_type_handle = H5Writer.fetch_handle(h5file, entity.entity_type)
```
Comment (sebhmg) [inline geoh5py/io/h5_writer.py:562]: **question:** is it per entity type? I was expecting it to be per "data type"

@jincandescent , do you confirm?

**Review by sebhmg — COMMENTED** [review]
waiting for @jincandescent to comment

File: geoh5py/io/h5_writer.py (hunk 1 above)
Comment (domfournier) [reply geoh5py/io/h5_writer.py:562]: here the entity is a "data" entity.

Comment (sebhmg) [reply geoh5py/io/h5_writer.py:562]: mmh... however the data type is not the `entity_type`, or is it?

Comment (sebhmg) [reply geoh5py/io/h5_writer.py:562]: data type is how the user regroups data for statistics. Typically done automatically based  on the name of the data.
@jincandescent  correct me if I am wrong

Comment (domfournier) [reply geoh5py/io/h5_writer.py:562]: Under geoh5py, the data_type is just a sub-class of entity_type, without a fix uuid
image

**Review by sebhmg — APPROVED** [review]

**Pull request merged**